src/MongoDB.Driver.Core/Core/Connections/BinaryConnection.cs

```diff
@@ -568,15 +589,17 @@ public async Task SendMessagesAsync(IEnumerable<RequestMessage> messages, Messag
             Ensure.IsNotNull(messages, nameof(messages));
             ThrowIfDisposedOrNotOpen();
 
-            var helper = new SendMessagesHelper(this, messages, messageEncoderSettings);
+            var helper = new SendMessagesHelper(this, messages, _sendCompressorType, _compressorSource, messageEncoderSettings);
             try
             {
                 helper.EncodingMessages();
-                using (var buffer = helper.EncodeMessages(cancellationToken))
+                var buffers = helper.EncodeMessages(cancellationToken);
+                using (var possiblyCompressedBuffer = buffers.Item1)
```

Comment:
As I wrote in another comment, I believe the code which is based on the new version of Tuple:

    var buffers = helper.EncodeMessages(cancellationToken);
    using (var possiblyCompressedBuffer = buffers.PossiblyCompressedBuffer)
    using (var originalMessageBuffer = buffers.OriginalMessageBuffer)

will look much better than current.